buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlRestTestPlugin.java

```diff
@@ -0,0 +1,122 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.test.rest;
+
+import org.elasticsearch.gradle.ElasticsearchJavaPlugin;
+import org.elasticsearch.gradle.VersionProperties;
+import org.elasticsearch.gradle.info.BuildParams;
+import org.elasticsearch.gradle.plugin.PluginPropertiesExtension;
+import org.elasticsearch.gradle.test.RestIntegTestTask;
+import org.elasticsearch.gradle.testclusters.RestTestRunnerTask;
+import org.elasticsearch.gradle.testclusters.TestClustersPlugin;
+import org.elasticsearch.gradle.util.GradleUtils;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaBasePlugin;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetContainer;
+import org.gradle.api.tasks.bundling.Zip;
+
+/**
+ * Apply this plugin to run the YAML based REST tests.
+ */
+public class YamlRestTestPlugin implements Plugin<Project> {
+
+    public static final String SOURCE_SET_NAME = "yamlRestTest";
+
+    @Override
+    public void apply(Project project) {
+
+        // yaml Rest tests require a Java test runner
+        project.getPluginManager().apply(ElasticsearchJavaPlugin.class);
+        // to spin up the external cluster
+        project.getPluginManager().apply(TestClustersPlugin.class);
+        // to copy around the yaml tests and json spec
+        project.getPluginManager().apply(RestResourcesPlugin.class);
+
+        // note - source sets are not created via org.elasticsearch.gradle.util.GradleUtils.addTestSourceSet since unlike normal tests
+        // we only want the yamlRestTestSourceSet on the classpath by default. The yaml tests should be pure black box testing over HTTP and
+        // such it should not need the main class on the class path. Also, there are some special setup steps unique to YAML REST tests.
+
+        // create source set
+        SourceSetContainer sourceSets = project.getExtensions().getByType(SourceSetContainer.class);
+        SourceSet yamlTestSourceSet = sourceSets.create(SOURCE_SET_NAME);
+
+        // create task - note can not use .register due to the work in RestIntegTestTask's constructor :(
+        // see: https://github.com/elastic/elasticsearch/issues/47804
+        RestIntegTestTask yamlRestTestTask = project.getTasks()
+            .create(
+                SOURCE_SET_NAME,
+                RestIntegTestTask.class,
+                task -> { task.dependsOn(project.getTasks().getByName("copyRestApiSpecsTask")); }
+            );
+        yamlRestTestTask.setGroup(JavaBasePlugin.VERIFICATION_GROUP);
+        yamlRestTestTask.setDescription("Runs the YAML based REST tests against an external cluster");
+
+        // setup task dependency
+        if (BuildParams.isInternal()) {
```

Comment:
The `compile` configuration is deprecated, please use "implementation" instead. Also, no need to construct the configuration name, you can request it via `SourceSet.getImplementationConfigurationName()`.

Reply:
done [`6289b00` (#56841)](https://github.com/elastic/elasticsearch/pull/56841/commits/6289b00b7fb71856968e2a6fa543202f4aa8cf7d) and [`02afa30` (#56841)](https://github.com/elastic/elasticsearch/pull/56841/commits/02afa305711b7118a64ced93df1607097a5defcb)